site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,353 @@
+</figure>
+
+Green lines indicate enough confidence to trigger prerendering. In this example typing "s" gives a reasonable confidence (amber), but once you type "sh" then Chrome has enough confidence that you nearly always navigate to `https://sheets.google.com`.
+
+This screenshot was taken in a relatively fresh Chrome install, and filtering out zero confidence predictions, but if you view your own predictors you will likely see considerably more entries, and potentially more characters required to reach a high enough confidence level.
+
+These predictors are also what drive the address bar suggested options you may have noticed:
+
+<figure>
+{% Img src="image/W3z1f5ZkBJSgL1V1IfloTIctbIF3/GcJd2GQ76qdXKbcUb1VD.png", alt="Screenshot of the address bar 'Typeahead' feature", width="400", height="216.5" %}
+</figure>
+
+Chrome will continually update its predictors based on your typing and selections.
+
+- For a greater than 50% confidence level (shown in amber), Chrome proactively preconnects to the domain, but does not prerender the page.
+- For a greater than 80% confidence level (shown in green), Chrome will prerender the URL.
+
+### Using the Speculation Rules API to prerender pages
+
+For the third prerender option, web developers can insert JSON instructions onto their pages to inform the browser about which URLs would benefit from prerendering:
+
+```html
+<script type="speculationrules">
+{
+  "prerender": [
+    {
+      "source": "list",
+      "urls": ["next.html", "next2.html"]
+    }
+  ]
+}
+</script>
+```
+
+The Speculation Rules API is planned to be expanded beyond this simple example with the addition of prefetch, [scores](https://github.com/WICG/nav-speculation/blob/main/triggers.md#scores) (for example, the likelihood of a navigation), and syntax to implement [document rules](https://github.com/WICG/nav-speculation/blob/main/triggers.md#document-rules) (for example, matching `href` patterns). For now, only the above syntax is supported in Chrome.
+
+For the initial launch in Chrome 108, prerender is restricted to same-origin pages, opened within the same tab. Chrome 109 plans expand this to allow [prerendering same-site cross-origin pages](https://chromestatus.com/feature/4899735257743360) (for example, `https://a.example.com` could prerender a page on `https://b.example.com`, where the other site has opted in to same-site, cross-origin prerendering). Future versions may also [enable prerendering in new tabs](https://bugs.chromium.org/p/chromium/issues/detail?id=1350676).
+
+#### Detecting of Speculation Rules API support
+
+You can feature detect Speculation Rules API support with standard HTML checks:
+
+```js
+if (HTMLScriptElement.supports && HTMLScriptElement.supports('speculationrules')) {
+  console.log('Your browser supports the Speculation Rules API.');
+}
+```
+
+Speculation rules can be:
```

Comment:
This paragraph and everything below it in this section is not really related to "Detecting of Speculation Rules API support". I think it'd make sense to move all this content under the higher-level "Using the Speculation Rules API to prerender pages" heading (or potentially just moving the "detection" bit last).

Reply:
Fair point. I added some headings so I could directly link to them (e.g. CSP conversation) but that now makes this "Detection" section in wrong place. Fixed.